File: src/main/c++/gameModeButtons.cpp
/*
*@description Main entry point of the videogame
*@author Regy Special
*@date (First version [Linux]) 2026 July 20th Sunday, 15:46:00
*@license GNU General Public License to stop private corporation to not share source code
*/
#include "../../include/main/c++/gameModeButtons.hpp"
gameModeButton::gameModeButton(dialog *root, const char *title, unsigned long color)
{
  this->root = root;
  this->title = title;
  this->text.color = color;

  this->background.color = 0;
  this->border.color = this->text.color;
  this->eventMask = ButtonPressMask | ExposureMask;
  this->dontPropagateMask = ButtonPress;
  this->graphicMask = GCForeground;

  XSetWindowAttributes gameModeButtonAttributes{
    .background_pixel = this->background.color,
    .border_pixel = this->border.color,
    .event_mask = this->eventMask,
    .do_not_propagate_mask = this->dontPropagateMask
  };
  this->id = XCreateWindow
  (
    this->root->root->display,
    this->root->id,
    this->x,
    this->y,
    this->width,
    this->height,
    this->border.width,
    this->root->root->screen->root_depth,
    InputOutput,
    this->root->root->screen->root_visual,
    this->attributeMask,
    &gameModeButtonAttributes
  );
  XGCValues gcValues = {
    .foreground = this->text.color
  };
  this->graphicId = XCreateGC(this->root->root->display, this->id, this->graphicMask, &gcValues);
  this->root->subWindows.push(this->id);
}
gameModeButton::~gameModeButton(){
  this->root->subWindows.remove(this->id);
  XFreeGC(this->root->root->display, this->graphicId);
}

int gameModeButton::show(unsigned int microseconds)
{
#if defined __WIN32 || defined __WIN64
  Sleep(microseconds);
  return ShowWinodw(this->id, SW_NORMAL);
#else
  usleep(microseconds);
  return XMapRaised(this->root->root->display, this->id);
#endif
}

int gameModeButton::hide(unsigned int microseconds)
{
#if defined __WIN32 || defined __WIN64
  Sleep(microseconds);
  return ShowWinodw(this->id, SW_HIDE);
#else
  usleep(microseconds);
  return XUnmapWindow(this->root->root->display, this->id);
#endif
}

File: src/main/swift/gameModeButtons.swift
/*
*@description Main entry point of the videogame
*@author Regy Special
*@date (First version [Linux]) 2026 July 20th Sunday, 15:46:00
*@license GNU General Public License to stop private corporation to not share source code
*/
#include "../../include/main/c++/gameModeButtons.hpp"
gameModeButton::gameModeButton(dialog*root,const char*title,unsigned long color){
  XSetWindowAttributes gameModeButtonAttributes{
    .background_pixel=0,
    .border_pixel=0x808080,
    .event_mask=(self.eventMask = ButtonPressMask|ExposureMask),
    .do_not_propagate_mask=(self.dontPropagateMask = ButtonPress)
  };
  self.root=root;
  self.title=title;
  self.id=XCreateWindow(
    self.root.root.display,
    self.root.id,
    self.x,
    self.y,
    self.width,
    self.height,
    self.border.width,
    self.root.root.visualInfo.depth,
    self.root.root.visualInfo.c_class,
    self.root.root.visualInfo.visual,
    self.attributeMask,
    &gameModeButtonAttributes
  );
  XGCValues gcValues={.foreground=(self.text.color=color)};
  self.graphicId=XCreateGC(self.root.root.display,self.id,(self.graphicMask=GCForeground),&gcValues);
}
gameModeButton::~gameModeButton(){
  XFreeGC(self.root.root.display,self.graphicId);
}